lib/firmware/fsm_msg_cosmos.h

```diff
@@ -0,0 +1,94 @@
+#include "keepkey/firmware/cosmos.h"
+
+void fsm_msgCosmosGetAddress(const CosmosGetAddress *msg)
+{
+    RESP_INIT(CosmosAddress);
+
+    CHECK_INITIALIZED
+
+    CHECK_PIN
+
+    HDNode *node = fsm_getDerivedNode(SECP256K1_NAME, msg->address_n, msg->address_n_count, NULL);
+    if (!node) { return; }
+    hdnode_fill_public_key(node);
+
+    cosmos_getAddress(node->public_key, resp->address);
+
+    if (msg->has_show_display && msg->show_display)
+    {
+        if (!confirm(ButtonRequestType_ButtonRequest_ProtectCall, _("Share public account ID?"), "%s", resp->address))
+        {
+            fsm_sendFailure(FailureType_Failure_ActionCancelled, NULL);
+            layoutHome();
+            return;
+        }
+    }
+
+    resp->has_address = true;
+
+    layoutHome();
+    msg_write(MessageType_MessageType_CosmosAddress, resp);
+}
+
+void fsm_msgCosmosSignTx(const CosmosSignTx *msg)
+{
+    CHECK_INITIALIZED
+    CHECK_PIN
+
+    HDNode *node = fsm_getDerivedNode(SECP256K1_NAME, msg->address_n, msg->address_n_count, NULL);
+    if (!node) { return; }
+    hdnode_fill_public_key(node);
+
+    // Confirm transaction basics
+    if (!confirm(ButtonRequestType_ButtonRequest_ProtectCall, _("Confirm Main Details"), "From: %s\nTo: %s\nAmount: %f ATOM", msg->from_address, msg->to_address, (float)msg->amount * 1E-6))
```

Comment:
It would be more user-friendly to show the account description than the `from_address`. For example: https://github.com/keepkey/keepkey-firmware/blob/695cbdc8fe04298179936dab0f81bf6a8701814a/lib/firmware/fsm_msg_coin.h#L217